packages/at_secondary_server/lib/src/verb/handler/pkam_verb_handler.dart

```diff
@@ -38,58 +39,113 @@ class PkamVerbHandler extends AbstractVerbHandler {
+    // authenticate if signature is valid
+    if (isValidSignature) {
+      atConnectionMetadata.isAuthenticated = true;
+      atConnectionMetadata.authType = pkamAuthType;
+      atConnectionMetadata.enrollmentId = enrollId;
+      response.data = 'success';
+    } else {
+      atConnectionMetadata.isAuthenticated = false;
+      logger.severe('pkam authentication failed');
+      throw UnAuthenticatedException('pkam authentication failed');
+    }
+  }
 
-    //retrieve stored secret using sessionid and atsign
-    var storedSecret = await keyStore.get('private:$sessionID$atSign');
-    storedSecret = storedSecret?.data;
+  @visibleForTesting
+  Future<ApkamVerificationResult> handleApkamVerification(
+      String enrollId, String atSign) async {
+    String enrollmentKey =
+        '$enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign';
+    var enrollData = await keyStore.get(enrollmentKey);
+    final atData = enrollData.data;
+    final enrollDataStoreValue =
+        EnrollDataStoreValue.fromJson(jsonDecode(atData));
+    EnrollStatus enrollStatus =
+        EnrollStatus.values.byName(enrollDataStoreValue.approval!.state);
+
+    ApkamVerificationResult apkamResult = ApkamVerificationResult();
+    apkamResult.response = _getApprovalStatus(
+        enrollStatus, enrollId, enrollDataStoreValue.approval!.state);
+    if (apkamResult.response.isError) {
+      return apkamResult;
+    }
+    apkamResult.publicKey = enrollDataStoreValue.apkamPublicKey;
+    return apkamResult;
+  }
 
+  Response _getApprovalStatus(
+      EnrollStatus enrollStatus, enrollId, approvalState) {
+    Response response = Response();
+    switch (enrollStatus) {
+      case EnrollStatus.denied:
+        response.isError = true;
+        response.errorCode = 'AT0025';
+        response.errorMessage = 'enrollment_id: $enrollId is denied';
+        break;
+      case EnrollStatus.pending:
+        response.isError = true;
+        response.errorCode = 'AT0026';
+        response.errorMessage = 'enrollment_id: $enrollId is pending';
+        break;
+      case EnrollStatus.approved:
+        // do nothing when enrollment is approved
+        break;
+      case EnrollStatus.revoked:
+        response.isError = true;
+        response.errorCode = 'AT0027';
+        response.errorMessage = 'enrollment_id: $enrollId is revoked';
+        break;
+      default:
+        response.isError = true;
+        response.errorCode = 'AT0026';
+        response.errorMessage =
+            'Could not fetch enrollment status for enrollment_id: $enrollId';
+        break;
+    }
+    return response;
+  }
+
+  Future<bool> _validateSignature(
+      var verbParams, var sessionId, String atSign, String publicKey) async {
+    var signature = verbParams[AT_PKAM_SIGNATURE]!;
+    var signingAlgo = verbParams[AT_PKAM_SIGNING_ALGO];
+    var hashingAlgo = verbParams[AT_PKAM_HASHING_ALGO];
     var signingAlgoEnum;
     var inputSignature;
```

Comment:
Please follow dart style and make this `Uint8List? inputSignature;`

Reply:
changed as part of [05f42a0](https://github.com/atsign-foundation/at_server/pull/1520/commits/05f42a080aa3b989ce7d6bddb7dbc3489b2825d5)